pkg/kubectl/resource/builder.go

```diff
@@ -32,6 +32,8 @@ import (
 	"k8s.io/kubernetes/pkg/util/errors"
 )
 
+var schemaFileExtensions = []string{".json", ".stdin", ".yaml", ".yml"}
```

Comment:
Sorry, phone won't let me paste, but its in bash_comp_utils.go where the filename flag is registered.  It adds a hint for locating files with the right extension.